torch/distributions/transforms.py

```diff
@@ -273,11 +273,14 @@ def log_abs_det_jacobian(self, x, y):
         if not self.parts:
             return torch.zeros_like(x)
         result = 0
-        for part in self.parts:
-            y = part(x)
-            result = result + _sum_rightmost(part.log_abs_det_jacobian(x, y),
+        for part in self.parts[:-1]:
+            y_tmp = part(x)
+            result = result + _sum_rightmost(part.log_abs_det_jacobian(x, y_tmp),
                                              self.event_dim - part.event_dim)
-            x = y
+            x = y_tmp
+        part = self.parts[-1]
+        result = result + _sum_rightmost(part.log_abs_det_jacobian(x, y),
```

Comment:
This is mainly to save the last transform computation, where we don't need to recompute `y`. It can help numerical stability *somehow* by using an exact `y` instead of an approximated `y` through transforms.